src/Time/StepChoosers/StepChooser.hpp

```diff
@@ -28,6 +32,41 @@ namespace StepChoosers {
 /// These can be passed in a list to the template argument of
 /// StepChooser to choose which StepChoosers can be constructed.
 namespace Registrars {}
+
+namespace detail {
+CREATE_HAS_TYPE_ALIAS(slab_choosers)
+CREATE_HAS_TYPE_ALIAS_V(slab_choosers)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(slab_choosers)
+CREATE_HAS_TYPE_ALIAS(step_choosers)
+CREATE_HAS_TYPE_ALIAS_V(step_choosers)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(step_choosers)
+CREATE_HAS_TYPE_ALIAS(compute_tags)
+CREATE_HAS_TYPE_ALIAS_V(compute_tags)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(compute_tags)
+CREATE_HAS_TYPE_ALIAS(simple_tags)
+CREATE_HAS_TYPE_ALIAS_V(simple_tags)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(simple_tags)
+}  // namespace detail
+
+template <typename Metavariables>
+using step_chooser_compute_tags =
+    tmpl::remove_duplicates<tmpl::flatten<tmpl::transform<
+        typename StepChooser<tmpl::remove_duplicates<tmpl::append<
+            detail::get_step_choosers_or_default_t<Metavariables, tmpl::list<>>,
+            detail::get_slab_choosers_or_default_t<
+                Metavariables, tmpl::list<>>>>>::creatable_classes,
+        tmpl::bind<detail::get_compute_tags_or_default_t, tmpl::_1,
```

Comment:
Heh, Will asked the same thing in his previous review. The answer after trying it was subtly, no, because the type-trait uses a `bool` with a default to do the template specialization logic, which breaks the brigand anticipation that all lambdas are templates with only types as parameters.

Reply:
Ah, fun... Okay, thanks for the explanation :)